torch/distributed/_fsdp/utils.py

```diff
@@ -0,0 +1,36 @@
+from typing import Dict, List, Tuple, Union, Any, Callable, Set
+from collections import OrderedDict  # noqa: F401
+
+import torch
+from torch.nn.utils.rnn import PackedSequence
+
+
+"""Useful functions to deal with tensor types with other python container types."""
+
+
+def _apply_to_tensors(fn: Callable, container: Union[torch.Tensor, Dict, List, Tuple, Set]) -> Any:
+    """Recursively apply to all tensor in different kinds of container types."""
+
+    def _apply(x: Union[torch.Tensor, Dict, List, Tuple, Set]) -> Any:
+        if torch.is_tensor(x):
+            return fn(x)
+        elif isinstance(x, OrderedDict):
```

Comment:
Do you need to handle OrderedDict here? All dictionaries are ordered in the python version we support.
If you want to keep it, then the functions signature types should be updated above.

Reply:
will update the signature